cmsplugin_feed_ai/feed.py

```diff
@@ -0,0 +1,190 @@
+        """
+        Initialize display item from raw feed object provided by
+        service specific API tool.
+        """
+        self.profile_name = profile_name
+
+    def __str__(self):
+        return "%s %s" % (self.source, self.text_content[:20])
+
+    def __repr__(self):
+        return "<%s {'source': %s, 'text_content': %s, 'created_at': %s}>" % (
+            type(self).__name__,
+            self.source,
+            self.text_content,
+            self.created_at
+        )
+
+    @property
+    def profile_url(self):
+        """
+        URL which points to the profile related to this feed item.
+        """
+        return "%s/%s" % (self.service_url, self.profile_name)
+
+    @property
+    def link(self):
+        return self.content_link or self.permalink
+
+
+class TwitterFeedDisplayItem(FeedDisplayItem):
+    source = "twitter"
+    service_url = "https://twitter.com"
+
+    def __init__(self, feed_obj, profile_name):
+        """
+        Initialize display item from raw Twitter feed object provided by
+        Twitter API client (Twython).
+        """
+        super(TwitterFeedDisplayItem, self).__init__(profile_name)
+        self.created_at = parse_datetime(feed_obj.get("created_at", ""))
+        self.html_content = Twython.html_for_tweet(feed_obj)
+        self.source_id = feed_obj.get("id_str", "")
+        self.text_content = feed_obj.get("text", "")
+
+    @property
+    def permalink(self):
+        """
+        Permalink which points to the original feed object in external service.
+        """
+        return "%s/status/%s" % (self.profile_url, self.source_id)
+
+
+class FacebookFeedDisplayItem(FeedDisplayItem):
+    source = "facebook"
+    service_url = "https://facebook.com"
+
+    def __init__(self, feed_obj, profile_name):
+        """
+        Initialize display item from raw facebook feed object provided by
+        facebook API client (facebook-sdk).
+        """
+        super(FacebookFeedDisplayItem, self).__init__(profile_name)
+        self.created_at = parse_datetime(feed_obj.get("created_time", ""))
+        self.content_link = feed_obj.get("link", "")
+        self.source_id = feed_obj.get("id", "")
+        self.text_content = feed_obj.get(
+            "message",
+            feed_obj.get("status_type", "").replace("_", " ").capitalize
+        )
+
+    @property
+    def permalink(self):
+        """
+        Permalink which points to the original feed object in external service.
+        """
+        return "%s/posts/%s" % (self.profile_url, self.source_id.split("_")[-1])
+
+
+def get_feed_settings(key=None, default=None):
+    feed_settings = getattr(settings, "AI_FEED_SETTINGS", {})
+    if key not in feed_settings:
+        warn("Missing %r setting in AI_FEED_SETTINGS" % key)
+        return default
+    return feed_settings.get(key, default)
+
+
+def get_feed_cache_key(service, profile_name, limit):
+    """Return a unique cache key for given service"""
```

Comment:
You use this function in exactly one place (and I foresee it won't be used elsewhere), so inline it.